src/trace/layers/hashed.rs

```diff
@@ -398,17 +403,17 @@ impl<K: HashOrdered+Clone+Default, L: MergeBuilder> HashedBuilder<K, L> {
 
 /// A cursor with a child cursor that is updated as we move.
 #[derive(Debug)]
-pub struct HashedCursor<K: HashOrdered, L: Cursor> {
+pub struct HashedCursor<B, K: HashOrdered, L: Cursor> {
 	shift: usize,			// amount by which to shift hashes.
 	bounds: (usize, usize),	// bounds of slice of self.keys.
 	pos: usize,				// <-- current cursor position.
 
-	keys: Rc<Vec<Entry<K>>>,
+	keys: OwningRef<Rc<B>, Vec<Entry<K>>>,
```

Comment:
Or, maybe this is a horrible mess if it creates a new generic parameter that must be everywhere present throughout lots and lots of types. Let it not be thought of as a uniformly great idea, but might want to think it out.